src/iop/watermark.c

```diff
@@ -615,17 +617,53 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
   const int ch = piece->colors;
   const float angle = (M_PI / 180) * (-data->rotate);
 
-  /* Load svg if not loaded */
-  gchar *svgdoc = _watermark_get_svgdoc(self, data, &piece->pipe->image);
-  if(!svgdoc)
+  gchar configdir[PATH_MAX] = { 0 };
+  gchar datadir[PATH_MAX] = { 0 };
+  gchar *filename;
+  dt_loc_get_datadir(datadir, sizeof(datadir));
+  dt_loc_get_user_config_dir(configdir, sizeof(configdir));
+  g_strlcat(datadir, "/watermarks/", sizeof(datadir));
+  g_strlcat(configdir, "/watermarks/", sizeof(configdir));
+  g_strlcat(datadir, data->filename, sizeof(datadir));
+  g_strlcat(configdir, data->filename, sizeof(configdir));
+
+  if(g_file_test(configdir, G_FILE_TEST_EXISTS))
+    filename = configdir;
+  else if(g_file_test(datadir, G_FILE_TEST_EXISTS))
+    filename = datadir;
+  else
+  {
+    dt_iop_image_copy_by_size(ovoid, ivoid, roi_out->width, roi_out->height, ch);
+    return;
+  }
+
+  // find out the watermark type
+  dt_iop_watermark_type_t type;
+  const gchar *extension = strrchr(data->filename, '.');
+  if(!g_ascii_strcasecmp(extension, ".svg"))
+    type = DT_WTM_SVG;
+  else if(!g_ascii_strcasecmp(extension, ".png"))
+    type = DT_WTM_PNG;
+  else // this should not happen
   {
     dt_iop_image_copy_by_size(ovoid, ivoid, roi_out->width, roi_out->height, ch);
     return;
   }
 
+  /* Load svg if not loaded */
+  gchar *svgdoc;
```

Comment:
`= NULL;` as could be not initialized, I understand that this will be used only if type = DT_WTM_SVG but better be clear/explicit for reviewer of the code in the future.

Reply:
done